### DISCUSSION TITLE
Get es futures historical data

### DISCUSSION THREAD

**Discussion by michael12987** [erdewit/ib_insync#562] [Q&A]
Hi, trying to get es contract historical data with no success. This is what i tried:

```
`contract = Future('ES','20230115','CME')

data = ib.reqHistoricalData(
        contract,
        endDateTime='',
        durationStr='1 Y',
        barSizeSetting='1 day',
        whatToShow='TRADES')
`

```
I'm not sure what am i doint wrong as i keep getting errors such as :
: No security definition has been found for the request, contract: Future(symbol='ES', lastTradeDateOrContractMonth='20230115', exchange='CME')

Thanks!

**Comment by jlixfeld** [comment]
```python
from ib_insync import IB, Forex, util, Future

util.startLoop()
ib = IB()
ib.connect(clientId=69)
contract = Future('ES','20230115','CME')
ib.qualifyContracts(contract)

Error 200, reqId 163: No security definition has been found for the request, contract: Future(symbol='ES', lastTradeDateOrContractMonth='20230115', exchange='CME')
Unknown contract: Future(symbol='ES', lastTradeDateOrContractMonth='20230115', exchange='CME')
```

So your contract definition is incorrect.

**Comment by August1328** [comment]
Since I was looking into using ib_insnyc with Futures in the last days, a comment: The historical data simply is not available, it is removed by IBKR once the options expire.

You can try for yourself:

`contract = Future('ES','202301','CME')` or `contract = Future('ES','202302','CME')` does result in this error.

`contract = Future('ES','202303','CME')` or including the correct day `contract = Future('ES','20230317','CME')` works.

See also a similar answer on Stackoverflow [https://stackoverflow.com/questions/56819763/no-security-definition-has-been-found-for-the-request-future-option-python](url)
Or directly from IBKR about Unavailable Historical Data [https://interactivebrokers.github.io/tws-api/historical_limitations.html#non-available_hd](url)
